src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -1861,14 +1884,21 @@ Status ColorControlServer::stepSaturationCommand(EndpointId endpoint, const Comm
  * @param commandData Struct containing the parameters of the command.
  * @return Status::Success when successful,
  *         Status::InvalidCommand when an unknown action or direction is provided
- *         Status::UnsupportedEndpoint when the provided endpoint doesn't correspond with a hue transition state.
+ *         Status::UnsupportedEndpoint when the provided endpoint doesn't correspond with a hue transition state,
+ *         Status::ConstraintError when a command parameters is outside its defined value range.
  */
 Status ColorControlServer::colorLoopCommand(EndpointId endpoint, const Commands::ColorLoopSet::DecodableType & commandData)
 {
     MATTER_TRACE_SCOPE("colorLoop", "ColorControl");
+    constexpr BitMask<ColorLoopUpdateFlags> maxUpdateFlagsValue(
+        ColorLoopUpdateFlags::kUpdateAction, ColorLoopUpdateFlags::kUpdateDirection, ColorLoopUpdateFlags::kUpdateTime,
+        ColorLoopUpdateFlags::kUpdateStartHue); // add any new bitmap field to this value!
+
     // Validate the action and direction parameters of the command
     VerifyOrReturnValue(commandData.action != ColorLoopActionEnum::kUnknownEnumValue, Status::InvalidCommand);
     VerifyOrReturnValue(commandData.direction != ColorLoopDirectionEnum::kUnknownEnumValue, Status::InvalidCommand);
+    // Command Parameters constraint checks:
+    VerifyOrReturnValue(commandData.updateFlags <= maxUpdateFlagsValue, Status::ConstraintError);
```

Comment:
Is this the right behavior, or should this have the usual "unknown bits are ignored" behavior?

Reply:
The spec didn't mention any reserved bits for possible future extension so I treated it like we do for enums but I don't have a strong opinion. I can remove this check. It is true that we don't do a constraint check on other bitmaps that I can think of